### DISCUSSION TITLE
Realtime should allow user's jwt access token to connect from web client

### DISCUSSION THREAD

**Discussion by kk21** [supabase/supabase#4086] [Feature Requests]
Hi,

When I used the user's jwt access token in web client's createClient() to subscribe to Realtime, I got repeated WebSocket connection failed errors.

However the subscription works using the anon or service key in createClient().
Using the anon key in createClient() and then setAuth(user's jwt) also works.

I wonder if using the user's jwt directly in createClient() should work so that the web client does not need the anon key?

Thanks.

**Comment by w3b6x9** [comment]
@kk21 I don't think it's possible at the moment to pass in the user's token directly into `createClient()` due to Supabase Auth. @kangmingtay has a better grasp of Auth so I'll leave it to him if he wants to provide additional details. For now, you'll have to either use the `anon` or `service_role` keys.
